### PR TITLE
README updates

### PR DESCRIPTION
**Pull request by CalMacCQ**
# Description

I realised I messed up some spacing for the manual and examples links in the README in #1284. Rather than just fix the spacing  I thought I would add the following edits.

* Added a brief explanation of what TKET is. Hopefully clarifying the difference between TKET and pytket.
* Reordered some subsections - moving the API docs links further up.
* Added badges for pypi release number, slack and stack exchange (can remove these if preferred).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,31 +1,53 @@
 # tket
 
+[![Slack](https://img.shields.io/badge/Slack-4A154B?style=for-the-badge&logo=slack&logoColor=white)](https://tketusers.slack.com/join/shared_invite/zt-18qmsamj9-UqQFVdkRzxnXCcKtcarLRA#/shared-invite/email)
+[![Stack Exchange](https://img.shields.io/badge/StackExchange-%23ffffff.svg?style=for-the-badge&logo=StackExchange)](https://quantumcomputing.stackexchange.com/search?q=user:19610+[pytket])
+[![PyPI version](https://badge.fury.io/py/pytket.svg)](https://badge.fury.io/py/pytket)
+
 ## Introduction
 
-This repository contains the full source code for tket, a quantum SDK.
+TKET (pronounced ticket) is a high performance quantum compiler that can optimise circuits for a wide range of quantum computing architectures.
```
Comment (cqc-alec) [inline README.md:9]:
```suggestion
TKET (pronounced "ticket") is a high-performance quantum compiler that can optimise circuits for a wide range of quantum computing architectures.
```

File: README.md [hunk 2]
```diff
@@ -1,31 +1,53 @@
 # tket
 
+[![Slack](https://img.shields.io/badge/Slack-4A154B?style=for-the-badge&logo=slack&logoColor=white)](https://tketusers.slack.com/join/shared_invite/zt-18qmsamj9-UqQFVdkRzxnXCcKtcarLRA#/shared-invite/email)
+[![Stack Exchange](https://img.shields.io/badge/StackExchange-%23ffffff.svg?style=for-the-badge&logo=StackExchange)](https://quantumcomputing.stackexchange.com/search?q=user:19610+[pytket])
```
Comment (cqc-alec) [inline README.md:4]: Should `user:19610` be part of this URL? When I follow the link I see a message about "not deleted user 19610" near the top of the search results and I don't know what this means.

File: README.md [hunk 3]
```diff
@@ -1,31 +1,53 @@
 # tket
 
+[![Slack](https://img.shields.io/badge/Slack-4A154B?style=for-the-badge&logo=slack&logoColor=white)](https://tketusers.slack.com/join/shared_invite/zt-18qmsamj9-UqQFVdkRzxnXCcKtcarLRA#/shared-invite/email)
```
Comment (cqc-alec) [inline README.md:3]: Is the last part of this URL (`#/shared-invite/email`) necessary? Seems to work without it.

**Review by cqc-alec — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (CalMacCQ) [reply README.md]: Ah this link got changed somehow.

Using `https://quantumcomputing.stackexchange.com/tags/pytket` seems to work

File: README.md (hunk 3 above)
Comment (CalMacCQ) [reply README.md]: I've removed this part. I don't think its needed.

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**